### DISCUSSION TITLE
ESP Proposal list incomplete/trimmed with IKEv1

### DISCUSSION THREAD

**Discussion by bhatnitish** [strongswan/strongswan#1678] [Q&A]
Hi,

I'm trying to bring up a IKEv1 tunnel between 2 nodes. This is my config on both ends - 

Node1 - 
connections {
    vpn1 {
      version = 1
      local_addrs = x.x.x.x
      remote_addrs = y.y.y.y
      proposals = aes256-sha512-ecp384
      rekey_time = 8h
      reauth_time = 20s
      dpd_delay = 5s
      dpd_timeout = 60s
      local {
          auth = psk
      }
      remote {
          auth = psk
      }
      children {
         vpn11 {
            esp_proposals = aes128gcm16-ecp256,aes256gcm16-ecp256,aes256gcm16-ecp384,aes128gcm16-ecp384
            local_ts = 0.0.0.0/0
            remote_ts = 0.0.0.0/0
            rekey_time = 4h
        }
     }
  }
}

Node2 - 
connections {
    vpn1 {
      version = 1
      local_addrs = y.y.y.y
      remote_addrs = x.x.x.x
      proposals = aes256-sha512-ecp384
      rekey_time = 8h
      dpd_delay = 5s
      dpd_timeout = 60s
      local {
          auth = psk
      }
      remote {
          auth = psk
      }
      children {
         vpn11 {
            esp_proposals = aes256gcm16-ecp256
            local_ts = 0.0.0.0/0
            remote_ts = 0.0.0.0/0
            rekey_time = 4h
            dpd_action = clear
        }
     }
  }
}

When I try to initiate the child from Node1, I see that the Child connection fails to come up. The ESP proposal list received by Node2 is incomplete. These are the CFG (level 2) Child specific logs on Node 2 - 

May  1 16:35:30 05[CFG2] <vpn1|86> looking for a child config for 0.0.0.0/0 === 0.0.0.0/0
May  1 16:35:30 05[CFG2] <vpn1|86> proposing traffic selectors for us:
May  1 16:35:30 05[CFG2] <vpn1|86>  0.0.0.0/0
May  1 16:35:30 05[CFG2] <vpn1|86> proposing traffic selectors for other:
May  1 16:35:30 05[CFG2] <vpn1|86>  0.0.0.0/0
May  1 16:35:30 05[CFG2] <vpn1|86>   candidate "vpn11" with prio 5+5
May  1 16:35:30 05[CFG2] <vpn1|86> found matching child config "vpn11" with prio 10
May  1 16:35:30 05[CFG2] <vpn1|86> selecting traffic selectors for other:
May  1 16:35:30 05[CFG2] <vpn1|86>  config: 0.0.0.0/0, received: 0.0.0.0/0 => match: 0.0.0.0/0
May  1 16:35:30 05[CFG2] <vpn1|86> selecting traffic selectors for us:
May  1 16:35:30 05[CFG2] <vpn1|86>  config: 0.0.0.0/0, received: 0.0.0.0/0 => match: 0.0.0.0/0
May  1 16:35:30 05[CFG2] <vpn1|86> selecting proposal:
May  1 16:35:30 05[CFG2] <vpn1|86>   no acceptable KEY_EXCHANGE_METHOD found
May  1 16:35:30 05[CFG2] <vpn1|86> selecting proposal:
May  1 16:35:30 05[CFG2] <vpn1|86>   no acceptable ENCRYPTION_ALGORITHM found
May  1 16:35:30 05[CFG1] <vpn1|86> received proposals: ESP:AES_GCM_16_256/ECP_384/NO_EXT_SEQ, ESP:AES_GCM_16_128/ECP_384/NO_EXT_SEQ
May  1 16:35:30 05[CFG1] <vpn1|86> configured proposals: ESP:AES_GCM_16_256/ECP_256/NO_EXT_SEQ
May  1 16:35:30 05[IKE1] <vpn1|86> no matching proposal found, sending NO_PROPOSAL_CHOSEN

Although AES_GCM_16_256/ECP_256 is a configured proposal on Node1, it's not being transmitted. This is only seen in Ikev1. Ikev2 initiates are fine, even if I choose to initiate the IKE and Child SAs separately. What could be the issue here?

**Comment by tobiasbrunner** [comment]
> I'm trying to bring up a IKEv1 tunnel between 2 nodes.

Never use IKEv1 between two strongSwan hosts (or at all). It's now [officially deprecated](https://datatracker.ietf.org/doc/html/rfc9395).

Anyway, since IKEv1 doesn't provide a proper way to reject the proposed DH group and select a different one during Quick Mode (unlike IKEv2's INVALID_KE_PAYLOAD notify), only proposals with the selected DH group will be sent. And if the DH group negotiated for IKE is also in at least one ESP proposal, that's the one that will be used (in your case `ecp384`). All other proposals will be discarded.
